src/crystal/project/git_dependency.cr

```diff
@@ -0,0 +1,60 @@
+require "crystal/project/dependency"
+require "crystal/project/project_error"
+
+module Crystal
+  class GitDependency < Dependency
+    getter target_dir
+
+    def initialize(@repo, name = nil : String?, branch = nil : String?)
+      unless repo =~ /(.*)(:|\/)(.*)\/(.*)(.git)/
```

Comment:
If we could find a way to manage the load path more dynamically, we wouldn't even need a name because we could actually skip symlinking it into libs and just add the clone to the load path.